src/textual/css/stylesheet.py

```diff
@@ -188,15 +188,16 @@ def read(self, filename: str) -> None:
             path = os.path.abspath(filename)
         except Exception as error:
             raise StylesheetError(f"unable to read {filename!r}; {error}")
-        self.source[path] = css
+        self.source[str(path)] = css
```

Comment:
to avoid disruption I simply cast the path to a `str` - so `self.source` is still a `dict[str, str]` :slightly_smiling_face: